TestProject/ClientApp/src/components/Header.js

```diff
@@ -1,22 +1,21 @@
 ﻿import React from 'react';
+import UserMenu from "./UserMenu";
 
-function Header() {
+function Header(props) {
     return (
         <header>
             <div className="row">
                 <nav className="col-12 navbar navbar-dark bg-dark">
                     <div className="container-fluid">
-                        <a className="navbar-brand">Test Project</a>
-                        <button onClick={OnClick} className="btn btn-light">Sign in</button>
+                        <a className="navbar-brand" href="/">Test Project</a>
+                        {props.name === undefined
```

Comment:
` === undefined` part is redundant here

Reply:
`name` is not the parameter that responsible for whether the user is authorized or not, you shouldn't rely on it here